### DISCUSSION TITLE
Responding to calendar (iCal/vCal) invitations

### DISCUSSION THREAD

**Discussion by shu8** [simple-login/app#1741] [Feature Requests]
Would it be possible to add a way to respond to calendar invitations (e.g., from Google Calendar, Microsoft Teams, etc.) sent to an alias?

Currently, I believe mail clients like Gmail and Outlook parse the `.ics`/`.vcs` attachments and display their own 'event' box with a Yes/No button or similar. I understand that it's probably not possible to change the behaviour of how these work because the RSVP is sent directly to the sender from the mail client and I don't think it goes through SimpleLogin.

However, maybe there could be an alternative way for such invitations to be RSVP'd through SimpleLogin?

I personally do not always want senders to know my main mailbox email address, and would rather they only know my alias email address (especially when using a custom domain).

---

I _think_ this could be implemented by checking on emails for an attachment that looks something like (mainly, `Content-Type: text/calendar`):

```
Content-Type: text/calendar; charset="utf-8"; method=REQUEST
Content-Transfer-Encoding: base64
```

and extracting the sender's email address, and constructing an RSVP message to send back (https://stackoverflow.com/questions/4430974/is-there-a-standard-and-an-implementation-for-icalendar-event-rsvp, using the `PARTSTAT` field defined in https://www.rfc-editor.org/rfc/rfc5545)

---

In terms of how this could be surfaced to users, a few options I can think of are:

- adding a section in the SimpleLogin dashboard/app UI (or even just the API) that allows RSVPs to be sent
  This could be as simple as just having a place to upload the `.ics`/`.vsc` attachment, automatically extract the event details and sender, and choose which alias you want to send an RSVP through
- users reply with a 'special' email content (e.g., a keyword followed by their YES/NO/MAYBE response) back to emails that contain these calendar attachments, which are detected at SimpleLogin and trigger an RSVP to be sent rather than the email itself

There may be more options too!

---

It would be great to hear others' thoughts on this. Personally, I think it could be really useful when using SimpleLogin not just for protecting privacy but also on custom domains for business/personal purposes (e.g., I believe the `@simplelogin.io` email addresses all use SimpleLogin internally for the aliases), as such calendar invites can be quite common!

I'd be more than happy to help contribute towards building this feature if it's thought to be reasonable and we decide on a general method! :) 

Note: I know Google Calendar invitations normally include some text and a link at the bottom which goes to a Google Calendar page where you can RSVP as the invited email address, but this is only for Google and not Microsoft which is still quite common!

**Comment by short4bmoney** [comment]
I don't know how complex/difficult a solution like the one you described would be, but I'm in favor of it.

However, if there's a simpler solution, so that invites sent to an alias can simply appear on (or be copied to) a Proton Calendar, that would be a good starting point.

Rather unfortunately, all the invites sent to aliases say _Your email address is not in the original participants list._ I understand the privacy-first mindset to not co-mingle data between services in order to maintain privacy, but if I **want** to use Proton Mail and Proton Calendar with Proton's acquisition of SimpleLogin, it should be more straightforward.

**Reply by colans** [reply]
If you want to see this implemented, go vote for [Allow adding a calendar invite to Proton Calendar when the email isn't in the participants list](https://protonmail.uservoice.com/forums/284483-proton-mail/suggestions/45483934-allow-adding-a-calendar-invite-to-proton-calendar).
